src/os/bluestore/BlueStore.cc

```diff
@@ -9968,9 +9968,7 @@ int BlueStore::_decompress(bufferlist& source, bufferlist* result)
     _set_compression_alert(false, alg_name);
     r = -EIO;
   } else {
-    if(alg == Compressor::COMP_ALG_ZLIB) {
-      cp->zlib_winsize = chdr.zlib_winsize;
-    }
+    cp->compressor_message = chdr.compressor_message;
```

Comment:
Suggest to add compressor_message parameter to compress/decompress functions instead of direct member access.
For the sake of encapsulation.

Reply:
Do you mean that I modify the compress/decompress functions for each compressor

Reply:
right, let's get/set this data through these functions